admin.py

```diff
@@ -15,13 +65,25 @@ def decorate(self, *args, **kwargs):
       kwargs: Parameters passed on to the specified function if successful.
     """
     user = users.get_current_user()
-    if not user:
-      logging.error('user is not logged in')
+    AbortIfUserIsNotLoggedIn(self, user)
+    AbortIfUserIsNotApplicationAdmin(self)
+
+    identifier = user.email()
+    if identifier is None or identifier is '':
+      logging.error('No identifier found for the user.')
+      self.abort(403)
+
+    user_is_admin = False
+    try:
+      directory_service = GoogleDirectoryService(OAUTH_DECORATOR)
+      user_is_admin = directory_service.IsAdminUser(identifier)
```

Comment:
nit: "user_is_admin" should rather be "is_admin_user" to be more consistent with boolean naming